test/e2e/runner/rpc.go

```diff
@@ -167,7 +167,11 @@ func waitForNode(ctx context.Context, node *e2e.Node, height int64) (*rpctypes.R
 				return nil, fmt.Errorf("timed out waiting for %v to reach height %v", node.Name, height)
 			case errors.Is(err, context.Canceled):
 				return nil, err
-			case err == nil && status.SyncInfo.LatestBlockHeight >= height:
+				// If the client is a LightClient, we want to query its trusted height
+			case err == nil && (node.Mode == e2e.ModeLight && status.LightClientInfo.LastTrustedHeight >= height):
+				return status, nil
```

Comment:
I'm not sure if this will work. `LastTrustedHeight` will never be updated unless `Update()` is called so essentially the light client will never catch up. 

I would recommend instead on line 131 you change it to `if node.Stateless() { return nil, nil } `

Reply:
Ah right I see that you modified the endpoint so that `/status` also calls the `/status` of the primary full node

Reply:
Actually, on line 174: the status of the full node is checked only if node.Mode is not ModeLight. So if the light client's LastTrustedHeight is not updated it would not catch up. But in the tests it is catching up. If we want the code to proceed even if the light client has not caught up, then I can just remove the check on 171 and leave the check for StateSync?

Reply:
I have committed a change to this. I am not returning nil if it is a light client (as e2e/start.go throws an exception). Instead I do return the status but do not wait for the light client to be at a particular height. This forced a change in start.go as well.